Dockerfile

```diff
@@ -1,9 +1,7 @@
-ARG PHP_VERSION=7.4
-ARG PHP_TARGET=php:${PHP_VERSION}-cli
+# add amd64 platform to support Mac M1
+FROM --platform=linux/amd64 shivammathur/node:latest-amd64
 
-FROM --platform=linux/amd64 ${PHP_TARGET}
-
-ARG COMPOSER_TARGET=2.0.3
+ARG PHP_VERSION=8.1
```

Comment:
We had another conversation about this, but does this now respect the PHP version? The value is hardcoded here, so I'm not sure if that's a default for an env var or if that's a fixed value

Reply:
>  but does this now respect the PHP version?

As discussed, the `shivammathur/node` installed PHP versions from `5.6` to `8.1`. And there is no way to specify the version using `node`. 
If we pass the argument `PHP_VERSION=8.0 docker-compose ...` looks like it does respect and set up the extensions for the given versions. 
But when I do `docker-compose exec -T test php -v "$@"` it always gives the `8.1`. 

Also, its argument, which available only for Building image.

Reply:
So what is the value used for exactly now? What does it affect?

Reply:
So, I spent some time on this and tried digging it more. 

> So what is the value used for exactly now? What does it affect?

We are using Argument for enabling extensions using `phpenmod`. Note that without specifying the PHP version, extensions do not enable. So, for now, the only purpose of the variable is to enable correct extensions. 

However, if want to enable the given php version we can do that too. 
I dug into how to use the same PHP version as the argument. Following commands, switch the given php version, and later down, extensions enable for the same version. 
```
# set PHP version
RUN sudo update-alternatives --set php /usr/bin/php$PHP_VERSION \
    && sudo update-alternatives --set phar /usr/bin/phar$PHP_VERSION \
    && sudo update-alternatives --set phar.phar /usr/bin/phar.phar$PHP_VERSION \
    && sudo update-alternatives --set phpize /usr/bin/phpize$PHP_VERSION \
    && sudo update-alternatives --set php-config /usr/bin/php-config$PHP_VERSION
```

Then we can set php version `PHP_VERSION=8.0 docker-compose build --no-cache && docker-compose up -d --force-recreate`

I tried the above code. Here is the `php -v` output in the container. 
<img width="474" alt="image" src="https://user-images.githubusercontent.com/54532330/186115311-dd1d7e1b-7cc1-4436-8cb9-2df83d164cb3.png">

Reply:
One more thing, As in this branch, we support only php `8.1`. If we run tests `./test`, I get the error. 
<img width="678" alt="image" src="https://user-images.githubusercontent.com/54532330/186116296-3492ec29-fdd5-4914-95a9-813657b8d3b7.png">

So what's the point of specifying the PHP version when we can't run the package below `8.1`? 

In fact, I was thinking of having a private variable (used only for extensions to avoid duplications) and stopping passing variables from outside.

Reply:
I see, so:
- the variable gets used for configuring PHP extensions
- in `ARG PHP_VERSION=8.1`, the `8.1` is the default, but it **does** respect outside environment variables
Correct?

If so, that's probably good and we can keep it that way.

And I'd still accept it as an environment variable, because even though *now* we are only using PHP 8.1, in the future we'll be running the tests on a matrix that includes PHP 8.2 and any newer versions.

Reply:
> the variable gets used for configuring PHP extensions

Yes, And we can use it for switching the PHP version. 

> in ARG PHP_VERSION=8.1, the 8.1 is the default, but it does respect outside environment variables

Yes, that's correct. 

> And I'd still accept it as an environment variable, because even though now we are only using PHP 8.1, in the future we'll be running the tests on a matrix that includes PHP 8.2 and any newer versions.

Sure. We can use the same variable for the environment as
```
ARG PHP_VERSION=8.1
ENV PHP_VERSION=${PHP_VERSION}
```

Reply:
> We can use the same variable for the environment as

Not sure what you mean by that code example. Doesn't the env var get automatically passed as the value for `ARG PHP_VERSION`?

Reply:
The 8.1 default should *still* be there. But we want to be able to override it using env vars.

Reply:
> The 8.1 default should still be there. But we want to be able to override it using env vars.

So should I add the PHP version switching code now? 

> Doesn't the env var get automatically passed as the value for ARG PHP_VERSION?

I don't think so. Because when I do `ENV PHP_VERSION=8.1`, passing PHP_VERSION variable from outside is not working. So `ARG` is working only.